AdminUi/apps/admin_ui/lib/home/identity_details/identity_details.dart

```diff
@@ -0,0 +1,256 @@
+import 'dart:io';
+
+import 'package:admin_api_sdk/admin_api_sdk.dart';
+import 'package:admin_api_types/admin_api_types.dart';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import 'package:intl/intl.dart';
+
+class IdentityDetails extends StatefulWidget {
+  final String address;
+
+  const IdentityDetails({required this.address, super.key});
+
+  @override
+  State<IdentityDetails> createState() => _IdentityDetailsState();
+}
+
+class _IdentityDetailsState extends State<IdentityDetails> {
+  Identity? _identityDetails;
+  late List<TierOverview>? _tiers;
+  String? _selectedTier;
+
+  late final ScrollController _scrollController;
+
+  @override
+  void initState() {
+    super.initState();
+    _scrollController = ScrollController();
+
+    _reloadIdentity();
+    _reloadTiers();
+  }
+
+  @override
+  void dispose() {
+    _scrollController.dispose();
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    if (_identityDetails == null) return const Center(child: CircularProgressIndicator());
+
+    final identityDetails = _identityDetails!;
+    return Scrollbar(
+      controller: _scrollController,
+      child: SingleChildScrollView(
+        controller: _scrollController,
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: [
+            if (Platform.isMacOS || Platform.isWindows) const BackButton(),
+            _IdentityDetailsCard(
+              identityDetails: identityDetails,
+              selectedTier: _selectedTier,
+              onTierChanged: (String? newValue) {
+                setState(() {
+                  _selectedTier = newValue;
+                });
+              },
+              availableTiers: _tiers!,
+              updateIdentity: _updateIdentity,
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+
+  Future<void> _updateIdentity() async {
+    if (_identityDetails == null) return;
+
+    final scaffoldMessenger = ScaffoldMessenger.of(context);
+
+    try {
+      await GetIt.I.get<AdminApiClient>().identities.updateIdentity(_identityDetails!.address, tierId: _selectedTier!);
+
+      scaffoldMessenger.showSnackBar(
+        const SnackBar(
+          content: Text('Identity updated successfully.'),
+          duration: Duration(seconds: 3),
+        ),
+      );
+
+      await _reloadIdentity();
+    } catch (e) {
+      scaffoldMessenger.showSnackBar(
+        const SnackBar(
+          content: Text('Failed to update identity. Please try again.'),
+          duration: Duration(seconds: 3),
+        ),
+      );
+    }
+  }
```

Comment:
This should be inside the tier switcher, and the only thing you update is the tier, so please call it `updateTierOfIdentity`